common/djangoapps/student/models/user.py

```diff
@@ -1240,6 +1251,13 @@ def log_successful_login(sender, request, user, **kwargs):  # lint-amnesty, pyli
 def log_successful_logout(sender, request, user, **kwargs):  # lint-amnesty, pylint: disable=unused-argument
     """Handler to log when logouts have occurred successfully."""
     if hasattr(request, 'user'):
+        tracker.emit(
+            USER_LOGGED_OUT_EVENT_NAME,
+            {
+                'user_id': user.id,
```

Comment:
Interesting that the code below uses `request.user.id`, I'm not sure if the test failures are related to that, but either way this may need to be wrapped to prevent those errors?

Reply:
Seems it's related. We can probably drop it as the user is added to the context

Reply:
We'd probably need to mock the tracker call as they do here: https://github.com/eduNEXT/edx-platform/blob/17b69a3f3a6677307f72032e6da79c793b34279e/common/djangoapps/student/management/tests/test_transfer_students.py#L92-L123
For those tests